### PR TITLE
[WIP] Get suspect IPs from ES now checks IPs from cloud providers

### PR DESCRIPTION
**Pull request by santinoncs**
Adding functionality in order to block IPs from specific providers and specific locations

Given this file

```
#ORG,CITY,COUNTRY,HOSTNAME
Google,Groningen,NL,bc.googleusercontent.com 
```

blocker will check lasts N min IPs from Elastic and will order based on given frequency and will block the ones that match the above info.

### CODE REVIEW

**Comment by rccrdpccl** [comment]
I wouldn't really go for this approach, as it would block only IPs that are causing enough 429s AND belong to specific ORG.

Wouldn't it be best to improve the flexibility for the accepted query, and allow the user to play with that?
That way any user can craft the query/queries to their needs.

Do you see any inconvenient with such approach?

**Comment by santinoncs** [comment]
> 

in fact there are 2 queries now, but it is a bit ugly maybe.
One query is the usual one ( 429..etc.)
and the other that queries everything in the lasts minutes, also configurable. It is below this second query that we are 
applying the blocking depending on country/city/hostname and org

How would you integrate that in just 1 query or maybe something cleaner?

**Comment by rccrdpccl** [comment]
How about having a list of queries passed through with all needed params?
Something like this:

```yaml
queries:
- query: "response.statusCode = 429"
  interval: 5m
  fields:
  - name: ip
    value: "geoip.ip"
- query: "geoip.Org = 'google' AND geoip.host = "myhost"
  interval: 25m
  fields:
  - name: ip
    value: "geoip.ip"
```

Of course this approach would not have the lookup for each IP (which could also be costly), and we should find a way to enrich this data in the database. Or we could think a way to inject some data enrichment at the IP blocker level.
What do you think?

**Comment by santinoncs** [comment]
> How about having a list of queries passed through with all needed params? Something like this:
> 
> ```yaml
> queries:
> - query: "response.statusCode = 429"
>   interval: 5m
>   fields:
>   - name: ip
>     value: "geoip.ip"
> - query: "geoip.Org = 'google' AND geoip.host = "myhost"
>   interval: 25m
>   fields:
>   - name: ip
>     value: "geoip.ip"
> ```
> 
> Of course this approach would not have the lookup for each IP (which could also be costly), and we should find a way to enrich this data in the database. Or we could think a way to inject some data enrichment at the IP blocker level. What do you think?

it is ok for me. Regarding the second query, this geoip.org data is not in ES.

**Comment by santinoncs** [comment]
@rccrdpccl @bvis 

First approach ( csv input file ) is not valid anymore.

File: charts/ip-blocker/values.yaml [hunk 1]
```diff
@@ -76,6 +85,52 @@ elasticSearch:
             }
           }
         }
+      queryElasticCityOrg.json: |
```
Comment (bvis) [inline charts/ip-blocker/values.yaml:88]: I'd extract this from here to just provide in the repo safe defaults or something useful but simpler.

**Review by bvis — APPROVED** [review]

**Pull request merged**